src/integration.test.ts

```diff
@@ -59,69 +91,110 @@ describe("Integration Test", (): void => {
     });
   });
 
-  const mockedExec = async (load: boolean, command: string): Promise<void> => {
-    const stdout = "standard output from Bash command";
-    const stderr = "error output from Bash command";
-    child_process.exec.mockImplementationOnce(
-      (_command: any, _options: any, callback: any): any => {
-        callback(null, { stdout, stderr });
-      }
+  const assertExecBashCommand = (
+    infoCallNum: number,
+    execCallNum: number,
+    command: string,
+    stdout: string
+  ): void => {
+    expect(core.info).nthCalledWith<[string]>(infoCallNum, command);
+    expect(child_process.exec).nthCalledWith<[string, ExecOptions, any]>(
+      execCallNum,
+      command,
+      EXEC_OPTIONS,
+      expect.anything()
     );
+    expect(core.info).nthCalledWith<[string]>(infoCallNum + 1, stdout);
+    expect(core.error).lastCalledWith(STDERR);
+    expect(core.setFailed).not.toHaveBeenCalled();
+  };
 
-    await (load ? docker.loadDockerImages() : docker.saveDockerImages());
-
+  const assertLoadDockerImages = (cacheHit: boolean): void => {
+    /* TODO: refactor to share code across test files to enable cleaner decomposition without
+     * failing copy/paste linter.
+     */
+    /* jscpd:ignore-start */
     expect(core.getInput).nthCalledWith<[string, InputOptions]>(1, "key", {
       required: true,
     });
-    expect(core.info).nthCalledWith<[string]>(1, command);
-    expect(child_process.exec).lastCalledWith(
-      command,
-      EXEC_OPTIONS,
-      expect.anything()
+    expect(cache.restoreCache).lastCalledWith([docker.DOCKER_IMAGES_PATH], KEY);
```

Comment:
Multiple instances: this was probably my own gaffe as a reviewer, but it just hit me that there are a lot of unnecessary asserts in the integration test associated with the mock implementations. I can see the point of asserting that a mock implementation wasn't called (or, in certain cases, that it was called `n` times), but by and large asserting what it was called with will be redundant assuming the mock implementation ensures that the test will fail when the mocks are called with the wrong arguments.